core/app/c/[communitySlug]/pubs/[pubId]/edit/page.tsx

```diff
@@ -0,0 +1,114 @@
+import type { Metadata } from "next";
+
+import Link from "next/link";
+import { notFound, redirect } from "next/navigation";
+
+import type { CommunitiesId, PubsId } from "db/public";
+import { Capabilities } from "db/src/public/Capabilities";
+import { MembershipType } from "db/src/public/MembershipType";
+import { Button } from "ui/button";
+
+import { ContentLayout } from "~/app/c/[communitySlug]/ContentLayout";
+import { PageTitleWithStatus } from "~/app/components/pubs/PubEditor/PageTitleWithStatus";
+import { PubEditor } from "~/app/components/pubs/PubEditor/PubEditor";
+import { getPageLoginData } from "~/lib/authentication/loginData";
+import { userCan } from "~/lib/authorization/capabilities";
+import { getCommunityBySlug } from "~/lib/db/queries";
+import { getPubsWithRelatedValuesAndChildren } from "~/lib/server";
+
+export async function generateMetadata({
```

Comment:
I mentioned something similar in a PR from kalil, but could you add this `React.cache`d version of the function like this, and use it in both the body and the metadata?

This way, the result of the entire function call is stored in memory and deduped during the rendering of the page.

Eg we won't have to read from the filesystem cache and compute the cache keys twice. This is somewhat of a limitation to `autoCache`s current api. See here for a more thorough explanation: https://github.com/pubpub/platform/pull/813#discussion_r1869676750 

Not a blocker, you can ignore it for now if you just want to get it in, but I thought i'd mention it so we're all aware of this!

```suggestion
const getPubsWithRelatedValuesAndChildrenCached = cache(
	async (pubId: PubsId, communityId: CommunitiesId) => {
		return getPubsWithRelatedValuesAndChildren({
			pubId,
			communityId,
		});
	}
);

```